File: Corrections/java/Neufplate/src/main/java/fr/neufplate/State/TitlingState.java
package fr.neufplate.State;

import fr.neufplate.CorporateBsClient;
import fr.neufplate.Neufplate;

public class TitlingState extends State {

    public TitlingState(Neufplate neufplate) {
        super(neufplate);
        //this.onTitling();
    }

    @Override
    public String onTitling() {
        String title = CorporateBsClient.generateCorporateBs();
        this.neufplate.nft.title = title;
        this.neufplate.changeState(new MakingCollisionState(this.neufplate));
        return title;
    }

    @Override
    public String onMakingCollision() {
        throw new RuntimeException("You must generate a title before making collision");
        //return null;
    }

    @Override
    public String onGenerating() {
        return null;
    }
}

File: Corrections/php/Neufplate/src/State/TitlingState.php
<?php

namespace App\State;

use App\CorporateBsClient;
use App\Event\EmailNotificationListener;
use App\Neufplate;

class TitlingState extends State
{


    public function __construct(Neufplate $neufplate)
    {
        parent::__construct($neufplate);
    }

    public function onTitling(): ?string
    {
        $title = CorporateBsClient::generateCorporateBs();
        $this->neufplate->nft->title = $title;
        $this->neufplate->changeState(new MakingCollisionState($this->neufplate));
        return $title;
    }

    public function onMakingCollision(): ?string
    {
        return null;
    }

    public function onGenerating(): ?string
    {
        return null;
    }
}